R/observed_data.R

```diff
@@ -968,16 +969,27 @@ reworked_figure <-
       }  
     }
     
+    print(" A B")
+    print(a)
+    print(b)
     if(isTRUE(date_constraint)){
-      if(as.Date(first(data$date)) > as.Date(last(data$date) - 40)){
+      print("DATE CONST")
+      print(as.Date(last(data$date)) - 40)
```

Comment:
Avoid hardcoding numbers. I noticed you passing the "interval_num" which is also 40 if these numbers are linked the variable should be used